src/modules/transaction/components/TxSignatureCollector/TxSignatureCollector.js

```diff
@@ -104,8 +106,12 @@ const TxSignatureCollector = ({
     });
   };
 
-  const onEnterPasswordSuccess = ({ privateKey }) =>
+  const onEnterPasswordSuccess = ({ privateKey }) => {
+    if (isTransactionAuthor) {
+      incrementNonce();
```

Comment:
the increment should happen once user clicks `Download`? otherwise following can happen

- The user intentionally closes the window without download, hence he has no file to sign, in that case the nonce should still be old one?
- The same applies for unintentionally close scenario as well?
- Now that we have switch scenario if the user clicks `Switch account` we might increase the nonce, so best to increment the nonce when user downloads the file or copies